src/services/dispatchers.ts

```diff
@@ -0,0 +1,111 @@
+
+    // Ensure all Dispatchers and actions have name fields
+    assertRequiredFields(localDispatcher);
+
+    // TODO: This does not account for dispatchers that exist in Extra Horizon with an existing name, but not a EXH_CLI_MANAGED tag
+    const exhDispatcher = exhDispatchers.find(({ name }) => name === localDispatcher.name);
+    await synchronizeDispatcher(sdk, localDispatcher, exhDispatcher);
+    console.groupEnd();
+  }
+}
+
+async function synchronizeDispatcher(sdk: OAuth1Client, localDispatcher: DispatcherCreation, exhDispatcher: Dispatcher | undefined) {
+  // Ensure all Dispatchers have the EXH_CLI_MANAGED tag
+  // eslint-disable-next-line no-param-reassign
+  localDispatcher.tags = localDispatcher.tags || [];
+  const hasCliManagedTag = localDispatcher.tags.includes(cliManagedTag);
+
+  if (!hasCliManagedTag) {
+    localDispatcher.tags.push(cliManagedTag);
+  }
+
+  if (!exhDispatcher) {
+    // A create Dispatcher request will also create actions
+    await dispatcherRepository.create(sdk, localDispatcher);
+  } else {
+    // Updating a Dispatcher will not update the actions, thus they must be synchronized separately
+    await dispatcherRepository.update(sdk, exhDispatcher.id, localDispatcher);
+    await synchronizeActions(sdk, localDispatcher, exhDispatcher);
+  }
+
+  console.log(green(`✅  Synchronized Dispatcher: ${localDispatcher.name}`));
+}
+
+async function synchronizeActions(sdk: OAuth1Client, localDispatcher: DispatcherCreation, exhDispatcher: Dispatcher) {
+  for (const localAction of localDispatcher.actions) {
+    console.group(blue(`Synchronizing Action: ${localAction.name}`));
+
+    // Find the corresponding Dispatcher Action in Extra Horizon
+    const exhAction = exhDispatcher.actions.find(({ name }) => name === localAction.name);
+
+    // Create or update Dispatcher Actions that exist locally
+    if (!exhAction) {
+      await dispatcherRepository.createAction(sdk, exhDispatcher.id, localAction);
+    } else {
+      await dispatcherRepository.updateAction(sdk, exhDispatcher.id, exhAction.id, localAction);
+    }
+
+    console.log(green(`✅  Synchronized Action: ${localAction.name}`));
+    console.groupEnd();
+  }
+
+  for (const exhAction of exhDispatcher.actions) {
+    // Find the corresponding Dispatcher Action in the local file
+    const actionExistsLocally = localDispatcher.actions.find(({ name }) => name === exhAction.name);
+
+    // Delete Dispatcher Actions that do not exist locally, but exist in Extra Horizon
+    if (!actionExistsLocally) {
+      console.group(blue(`Deleting Action: ${exhAction.name}`));
+
+      await dispatcherRepository.deleteAction(sdk, exhDispatcher.id, exhAction.id);
+
+      console.log(green(`✅  Synchronized Action: ${exhAction.name}`));
+      console.groupEnd();
+    }
+  }
+}
+
+async function extractDispatchersFromFile(file: string) {
+  const data = await readFile(file);
+  const dispatchers: DispatcherCreation[] = JSON.parse(data.toString());
```

Comment:
I feel like we might want to add some nice error handling here for if the parsing fails?

Reply:
I also feel like typing this already as `DispatcherCreation[]` is a bit of a stretch since there can be any json in there?  Should we do the validation here already, so that we can be more confident about the type we return.

Reply:
Oy, it's actually there to type the result of the function implicitly because otherwise it's just a Promise<any> but I can declare the response type explicitly? 

What error handling did you have in mind? Just a basic catch on the read file and throw a "An error occured parsing <path>" or something?